source/frontend/src/features/mapSideBar/property/tabs/propertyDetailsManagement/activity/edit/PropertyActivityEditContainer.tsx

```diff
@@ -0,0 +1,192 @@
+import { useCallback, useContext, useEffect, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+
+import { useOrganizationRepository } from '@/features/contacts/repositories/useOrganizationRepository';
+import { usePersonRepository } from '@/features/contacts/repositories/usePersonRepository';
+import { SideBarContext } from '@/features/mapSideBar/context/sidebarContext';
+import { usePropertyActivityRepository } from '@/hooks/repositories/usePropertyActivityRepository';
+import {
+  Api_PropertyActivity,
+  Api_PropertyActivityInvolvedParty,
+  Api_PropertyActivitySubtype,
+  Api_PropertyMinistryContact,
+} from '@/models/api/PropertyActivity';
+import { SystemConstants, useSystemConstants } from '@/store/slices/systemConstants';
+
+import { IPropertyActivityEditFormProps } from './PropertyActivityEditForm';
+
+export interface IPropertyActivityEditContainerProps {
+  propertyId: number;
+  propertyActivityId?: number;
+  onClose: () => void;
+  View: React.FunctionComponent<React.PropsWithChildren<IPropertyActivityEditFormProps>>;
+}
+
+export const PropertyActivityEditContainer: React.FunctionComponent<
+  React.PropsWithChildren<IPropertyActivityEditContainerProps>
+> = ({ propertyId, propertyActivityId, onClose, View }) => {
+  const { getSystemConstant } = useSystemConstants();
+
+  const history = useHistory();
+
+  const { setStaleLastUpdatedBy } = useContext(SideBarContext);
+
+  const [show, setShow] = useState(true);
+
+  const [loadedActivity, setLoadedActivity] = useState<Api_PropertyActivity | undefined>();
+
+  const [subtypes, setSubtypes] = useState<Api_PropertyActivitySubtype[]>([]);
+
+  const {
+    getActivitySubtypes: { execute: getSubtypes, loading: getSubtypesLoading },
+    getActivity: { execute: getActivity, loading: getActivityLoading },
+    createActivity: { execute: createActivity, loading: createActivityLoading },
+    updateActivity: { execute: updateActivity, loading: updateActivityLoading },
+  } = usePropertyActivityRepository();
+
+  const {
+    getPersonDetail: { execute: getPerson, loading: loadingPerson },
+  } = usePersonRepository();
+
+  const {
+    getOrganizationDetail: { execute: getOrganization, loading: loadingOrganization },
+  } = useOrganizationRepository();
+
+  // Load the subtypes
+  const fetchSubtypes = useCallback(async () => {
+    const retrieved = await getSubtypes();
+    if (retrieved !== undefined) {
+      setSubtypes(retrieved);
+    } else {
+      setSubtypes([]);
+    }
+  }, [getSubtypes]);
+
+  useEffect(() => {
+    fetchSubtypes();
+  }, [fetchSubtypes]);
+
+  const fetchMinistryContacs = useCallback(
```

Comment:
some potential to extract these into a hook that the variants of this form can use, would reduce code duplication and therefore the desire to edit a change in multiple places if fixes to this logic were required.